azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageTypeServlet.java

```diff
@@ -74,10 +82,36 @@ protected void handleGet(
       throws ServletException, IOException {
     try {
       final Map<String, String> templateVariableToValue = new HashMap<>();
-      if (IMAGE_TYPE_WITH_ID_URI_TEMPLATE.match(req.getRequestURI(), templateVariableToValue)) {
+      ImageTypeDTO imageTypeDTO = null;
+      if (StringUtils.isNotBlank(req.getPathInfo())) {
+        String imageTypeName = req.getPathInfo().substring(INDEX_AFTER_LEADING_SLASH);
+        if (imageTypeName.contains(PATH_SEPARATOR)) {
+          throw new ImageMgmtInvalidInputException(
+              "The path provided is not supported by this API. Please check the documentation.");
+        }
+        if (hasImageManagementPermission(imageTypeName, session.getUser(), Type.GET)) {
+          imageTypeDTO = this.imageTypeService.findImageTypeWithOwnersByName(imageTypeName);
+        } else if (hasParam(req, "proxyUser")) {
```

Comment:
Is a similar support for proxyUsers required for other parts of the ImageManagement APIs as well or is current set of changes enough for Jenkins/CRT needs ?

Reply:
I don't believe this is required for anything else, but I think there is hope that this is value added for ImageManagement Users.

Reply:
Wondering if something similar is needed for updating rampup plans ? (Please ignore, if that's not in the scope for this work)

Reply:
@sshardool We are adding this API to validate the proxyUser has permission on any image type. We are validating that using this API from our CICD and then allowing to call any other APIs like create/update/delete ramp up plan or create image version.

Reply:
Thanks, that clarifies things. Looking at the PR I got the impression that we are attempting to add impersonation as part of the API itself, hence the multiple questions. I believe a more accurate description is that its updating the existing `imageType` API endpoint to return information about users other than the currently logged in user (i.e the owner of session-id). Which seems fine as long as the executing user (with session-id) has privileges for doing this.